proposals/csharp-9.0/nullable-reference-types-specification.md

```diff
@@ -285,62 +307,154 @@ If a `simple_name` is not classified as a value, its null state is "not null". O
+A `null_conditional_expression` has the null state based on the properties of expression type:
+
+- If the type is a known value type then it has the null state "maybe null"
+- Else if the type is a type parameter then it has the null state "maybe default"
+- Else it has the null state "maybe null"
 
 ### Cast expressions
 
-If a cast expression `(T)E` invokes a user-defined conversion, then the null state of the expression is the default null state for its type. Otherwise, if `T` is *nullable* then the null state is "maybe null". Otherwise the null state is the same as the null state of `E`.
+If a cast expression `(T)E` invokes a user-defined conversion, then the null state of the expression is the default null state for the type of the user-defined conversion. Otherwise:
 
-***This needs upddating***
+- If `T` is a *nullable* type in the form `U?` where `U` is a type parameter then the null state is "maybe default"
+- Else If `T` is a *nullable* type then the null state is "maybe null"
+- Else the null state is the same as the null state of `E`
+
+### Unary and binary operators
+
+If a unary or binary operator invokes an user-defined operator that is declared with one or more attributes for special null behavior, the null state is determined by those attributes. Otherwise the null state of the expression is the default null state of its type.
+
+***Something special to do for binary `+` over strings and delegates?***
 
 ### Await expressions
 
 The null state of `await E` is the default null state of its type.
 
 ### The `as` operator
 
-An `as` expression has the null state "maybe null".
+The null state of an `E as T` expression depends first on properties of the type `T`. If the type *nonnullable* then the null state is "not null". Otherwise the null state depends on the conversion from the type of `E` to type `T`:
+
+- If the conversion Identity, boxing, implicit reference or implicit nullable has the same null state as `E`
+- Else if `T` is a type parameter then it has the null state "maybe default"
+- Else it has the null state "maybe null"
 
 ### The null-coalescing operator
 
-`E1 ?? E2` has the same null state as `E2`
+The null state of `E1 ?? E2` is the null state of `E2`
 
 ### The conditional operator
 
-The null state of `E1 ? E2 : E3` is "not null" if the null state of both `E2` and `E3` are "not null". Otherwise it is "maybe null".
+The null state of `E1 ? E2 : E3` is based on the null state of `E2` and `E3`:
+
+- If both are "not null" then the null state is "not null"
+- Else if either is "maybe default" then the null state is "maybe default"
+- Else the null state is "not null"
 
 ### Query expressions
 
 The null state of a query expression is the default null state of its type.
 
+*Additional work needed here*
+
 ### Assignment operators
 
 `E1 = E2` and `E1 op= E2` have the same null state as `E2` after any implicit conversions have been applied.
 
-### Unary and binary operators
```

Comment:
Just moved this closer to the cast section as I think they are going to have similar language when we are done here.